src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -399,7 +399,7 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                 match tcx.type_of(did).kind() {
                     ty::Adt(def, _) if def.is_enum() => {
                         if let Some(field) =
-                            def.all_fields().find(|f| f.ident.name == variant_field_name)
+                            def.all_fields().find(|f| f.ident(tcx).name == variant_field_name)
```

Comment:
There's no need for this to call `.ident(tcx)` AFAICT; it actually wants a Symbol.